src/routes/guides/testing-your-app.mdx

```diff
@@ -0,0 +1,404 @@
+      Tests  [1;32m1 passed[0;8m (1)
+   Start at  [1;1m16:51:19[0;8m
+   Duration  [1;1m4.34s[0;8m (transform 1.01s, setup 205ms, collect 1.54s, tests 155ms,
+environment 880ms, prepare 212ms)
+
+```
+
+### Rendering the component
+
+`render` from `@solidjs/testing-library` sets up a container, renders the component within and registers it for cleanup so that the container will removed after the successful test. It also handles wrapping the component in contexts and set up a router.
+
+```tsx frame="none"
+const renderResult = render(
+  () => <MyComponent />, // @solidjs/testing-library requires a function
+  { // all options are optional
+    container, // manually set up your own container, will not be handled
+    baseElement, // parent of container in case it is not supplied
+    queries, // manually set up custom queries
+    hydrate, // set to `true` to use hydration
+    wrapper, // reusable wrapper component to supply context
+    location, // sets up a router pointed to the location if provided
+  }
+)
+const {
+  asFragment, // function returning the contents of the container
+  baseElement, // the parent of the container
+  container, // the container in which the component is rendered
+  debug, // a function giving some helpful debugging output
+  unmount, // manually removing the component from the container
+  ...queries, // functions to select elements from the container
+} = renderResult
+```
+
+#### Using the right queries
+
+```
+                                          ⎧ Role
+                              get ⎫  By   ⎪ DisplayValue
+                            query ⎬       ⎨ LabelText
+                             find ⎭ AllBy ⎪ Text
+                                          ⎩ ...
+```
+
+The first and second part depends on if the query should wait for the element to (not) appear and if it should throw if the element cannot be found:
+
+- **getBy**: synchronous, throws if not found or more than 1 matches
+- **getAllBy**: synchronous, throws if not found, returns array of matches
+- **queryBy**: synchronous, null if not found, error if more than 1 matches
+- **queryAllBy**: synchronous, returns array of zero or more matches
+- **findBy**: asynchronous, rejected if not found within 1000ms or more than 1 matches, resolves wth element if found
+- **findAllBy**: asynchronous, rejected if not found within 1000ms, resolves with array of one or more element(s)
+
+By default, you should use a query starting with `get...`. Use `getAllBy...` if there are multiple elements matching the same query, otherwise use `getBy...`.
+
+There are two exceptions:
+
+1. if the `location` option is used or the component is based on resources, the router will be lazy-loaded, thus the first query after rendering needs to be `find...`
+2. If it should be tested whether something is *not* rendered, first find something that will be rendered at the same time and then use `query...` to test if the result is null or an empty array.
```

Comment:
I reduced that to using the empty array, since then we will get the elements nicely formatted in the error message. Otherwise, we would get the error from expect in case it was one element and the one from the query if there were multiple elements; consistency is king.